lib/dialects/abstract/query-generator.js

```diff
@@ -300,10 +300,21 @@ class QueryGenerator {
     }
 
     if (this._dialect.supports.inserts.updateOnDuplicate && options.updateOnDuplicate) {
-      onDuplicateKeyUpdate = ` ON DUPLICATE KEY UPDATE ${options.updateOnDuplicate.map(attr => {
-        const key = this.quoteIdentifier(attr);
-        return `${key}=VALUES(${key})`;
-      }).join(',')}`;
+      if (this._dialect.supports.inserts.updateOnDuplicate == ' ON CONFLICT DO UPDATE SET') { // postgres
```

Comment:
Replace this with (modify if anything missing)

```js
if (this._dialect.supports.inserts.updateOnDuplicate == ' ON CONFLICT DO UPDATE SET') { // postgres
  // If no conflict target columns were specified, use the primary key names from options.upsertKeys
  const conflictKeys = options.upsertKeys.map(attr => this.quoteIdentifier(attr));
  const updateKeys = options.updateOnDuplicate.map(attr => `${this.quoteIdentifier(attr)}=EXCLUDED.${this.quoteIdentifier(attr)}`);
  onDuplicateKeyUpdate = ` ON CONFLICT (${conflictKeys.join(',')}) DO UPDATE SET ${updateKeys.join(',')}`;
} else { // mysql / maria
  const valueKeys = options.updateOnDuplicate.map(attr => `${this.quoteIdentifier(attr)}=VALUES(${this.quoteIdentifier(attr)})`);
  onDuplicateKeyUpdate = `${this._dialect.supports.inserts.updateOnDuplicate} ${valueKeys.join(',')}`;
}
```

Reply:
Much cleaner, thanks for the coding tips :+1: